### DISCUSSION TITLE
[ssr] createRenderRoot return this supported?

### DISCUSSION THREAD

**Discussion by fetishcoder** [lit/lit#3416] [Q&A]
Hi all,

I am trying out the lab-ssr lib. very cool :)

But when testing an element with  
`createRenderRoot(){ return this }`
I can see its render output being added to the shadow dom.

Is this not supported yet?

I looked at 
https://github.com/lit/lit/blob/main/packages/lit-element/src/experimental-hydrate-support.ts
but dont really understand what is going on there :)

cheers

**Comment by augustjk** [comment]
Indeed, it is not supported. More explanation here: https://github.com/lit/lit/issues/3080

**Reply by justinfagnani** [reply]
I have this draft PR we need to decide what to do with: https://github.com/lit/lit/pull/3083

**Reply by fetishcoder** [reply]
Thanks for the quick replies.

Looking at 3083 I see:
```
if (this.element.renderRoot !== getShadowRoot(this.element)) {
      throw new Error('Cannot render custom render roots');
 }
```

So, can i conclude from this there is no plan to support this at all?
Or is there a glimmer of hope...

fyi I use `createRenderRoot(){ return this }` for a couple of reasons:
1. As a trick to allow a form tag to be able to see the input tags in a custom-component (as suggested per documentation)
2. As a declarative leafnode template call -- for a html fragment that needs to share the page level css

**Comment by fetishcoder** [comment]
Hello,
I was still trying to see if I could get this to work.
And I was able to make something work.

https://github.com/lit/lit/commit/e81f757fa2f006cd3f4c3124a0c83fc029b2d030

There are a few issues with it.
But it doesn't seem impossible.

What do you guys think?

**Comment by sepetboy** [comment]
您好，您的邮件已收到，我会尽快查收！
